### PR TITLE
Visitor wants to know when they are put on hold

### PR DESCRIPTION
**Pull request by kuldar-daniel**
This feature allows visitor to see when the are put on hold. The cue is:
* In Call view the "on hold" text appears under operator name
* In Call view the bottom text lets visitor know they can continue browsing while on hold
* In Call remote video stream is hidden and operator image is shown again
* In Call view the the operator image displays blurred overlay with on hold icon
* In Minimzed Bubble the same blurred overlay is also displayed

MUIC-241

### CODE REVIEW

File: GliaWidgets/Component/Connect/Operator/ConnectOperatorView.swift
Comment (igorkravchenko) [inline GliaWidgets/Component/Connect/Operator/ConnectOperatorView.swift:106]: Let's move image size to dedicated constant.

File: GliaWidgets/View/Call/CallView.swift
Comment (igorkravchenko) [inline GliaWidgets/View/Call/CallView.swift:26]: What happens for `else` case? Does `secondLabel.text` and  `connectView.statusView.setSecondText` still has  text assigned?

File: GliaWidgets/View/Call/CallView.swift
Comment (igorkravchenko) [inline GliaWidgets/View/Call/CallView.swift:28]: Do we to need address other modes along with `.video`?

File: GliaWidgets/View/Call/CallView.swift
Comment (igorkravchenko) [inline GliaWidgets/View/Call/CallView.swift:31]: Is it possible to add test that will ensure that all theses flags are flipped as expected?

File: GliaWidgets/View/Call/CallView.swift
Comment (igorkravchenko) [inline GliaWidgets/View/Call/CallView.swift:38]: Also handle this case in tests as well?

File: GliaWidgets/ViewModel/Call/CallViewModel.swift
Comment (igorkravchenko) [inline GliaWidgets/ViewModel/Call/CallViewModel.swift:103]: ViewModels seem to be like a good candidate for unit testing. Let's discuss how we can add tests here. Seems like `voice-over` branch has a lot of helpful mocks and Environments ready to be uses for testing.

File: GliaWidgets/View/Call/CallView.swift
Comment (kuldar-daniel) [reply GliaWidgets/View/Call/CallView.swift:28]: It only applies to video since the video mode has the connectview hidden and remote video stream visible.

File: GliaWidgets/View/Call/CallView.swift
Comment (kuldar-daniel) [reply GliaWidgets/View/Call/CallView.swift:29]: added else clause. It wasn't strictly needed due to the duration text would then be shown, but for claritys sake I added the duration text on the else

File: GliaWidgets/Component/Connect/Operator/ConnectOperatorView.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/Connect/Operator/ConnectOperatorView.swift:106]: the size varies depending whether it is in bubble or in call view though

File: GliaWidgets/Component/Bubble/BubbleStyle.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/Bubble/BubbleStyle.swift:10]: Perhaps I miss something, however I'd like to clarify why `onHoldOverlay` is optional and is variable?
For me, it looks like if this style is nil then `bubble` view will not show on hold state at all - [code](https://github.com/salemove/ios-sdk-widgets/pull/248/files#diff-811f3bc73367f56c237557e63c8e38abab3438cf50662d4dfea7ac21da63c7b7R59).
If we would like to apply default value, I think it should be some default `OnHoldOverlayStyle` and it should not lead to incorrect appearance of Bubble view.
Does it make sense?

File: GliaWidgets/Component/Bubble/BubbleView.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/Bubble/BubbleView.swift:58]: Since `show/hide` methods are not synced and we don't have tests for it, I'd like to propose to call `onHoldView?.removeFromSuperview()` on the beginning of this method, to be sure that after accidentally called `showOnHoldView` we will not have an additional `onHoldView` subview.

File: GliaWidgets/Component/Bubble/BubbleView.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/Bubble/BubbleView.swift:65]: does it make sense to put this size value inside of style?

File: GliaWidgets/Component/Connect/Operator/ConnectOperatorStyle.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/Connect/Operator/ConnectOperatorStyle.swift:18]: looks like the doc is missed here.

File: GliaWidgets/Component/Connect/Operator/ConnectOperatorView.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/Connect/Operator/ConnectOperatorView.swift:102]: I'd like to propose to call on the beginning of this method `onHoldView?.removeFromSuperview()`. To be sure that accidentally called `showOnHoldView` will not increment amount of onHoldView subviews.

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift:3]: Since we introduce a new style, I'd like to propose to avoid extending the global namespace and introduce this structure inside of Theme namespace for example.
What do you think, does it make sense?
cc: @igorkravchenko

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift:2]: let's add documentation for this structure.

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayView.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayView.swift:35]: perhaps I miss something, but for me it looks like `blurEffectView` will be placed under the imageView and I don't see any re-ordering here. What is the benefit of this `blurEffectView` view?

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift
Comment (dukhovnyi) [inline GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift:21]: why do we need to stop animation explicitly on deinitialization? will it cause any issues if we will deinit it without stopping the animator?

File: GliaWidgets/View/Call/CallStyle.swift
Comment (dukhovnyi) [inline GliaWidgets/View/Call/CallStyle.swift:49]: Why do we need to have these two strings optional? If these strings are nil visitors can miss context what happened with UI.

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift:3]: This is consitent to what we have with other components. While I don't mind it being in Theme generally, I think consistency would win here until we would refactor the themeing

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayView.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayView.swift:35]: imageView is for the pause icon asset. The whole overlay view will be placed on top of the user image view, which will blur the user image, but not the pause icon asset on top of it

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift:13]: This is used to make the intensity of default iOS blur a little less to achieve the desired effect in the design. Default blureffectview would make the imageview that is blurred very odd and this seemed like a better solution than to make my own blurring of image

File: GliaWidgets/View/Call/CallStyle.swift
Comment (kuldar-daniel) [reply GliaWidgets/View/Call/CallStyle.swift:49]: this was done due to backward compatiblity. I saw that similar approach was used previously while developing new features. But default values are actually quite good alternative tht you proposed and I didn't think of it myself. WIll fix

File: GliaWidgets/Component/Bubble/BubbleView.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/Bubble/BubbleView.swift:65]: That could work.

File: GliaWidgets/Component/Bubble/BubbleStyle.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/Bubble/BubbleStyle.swift:10]: I think the same reason why badge style is optional. It was developed after and we didn't want to force integrators to implement them with errors. But default value is very good idea.

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift
Comment (dukhovnyi) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift:13]: so, it is expected that we create an animator but don't use it, or do we start animation in some other place?

File: GliaWidgets/View/Call/CallStyle.swift
Comment (dukhovnyi) [inline GliaWidgets/View/Call/CallStyle.swift:96]: Since we have a pretty big constructor already, perhaps it makes sense to create a separate structure `OnHold` and place there two properties for now?

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift:21]: Yes, in case of incorrect memory usage where the view would stay alive (eg. retain cycle), this at least ensures that animation will be ended properly instead of running and wasting resources forever

File: GliaWidgets/View/Call/CallView.swift
Comment (kuldar-daniel) [reply GliaWidgets/View/Call/CallView.swift:39]: Added tests

File: GliaWidgets/View/Call/CallView.swift
Comment (kuldar-daniel) [reply GliaWidgets/View/Call/CallView.swift:46]: Added tests in fdeef7bf8b803a5f72352a760db62aa5ef2d6737

File: GliaWidgets/ViewModel/Call/CallViewModel.swift
Comment (kuldar-daniel) [reply GliaWidgets/ViewModel/Call/CallViewModel.swift:57]: Added tests in 80bbe14e21ed66104d3535270a2bbaadb4c11327

File: GliaWidgets/Component/Bubble/BubbleView.swift
Comment (kuldar-daniel) [reply GliaWidgets/Component/Bubble/BubbleView.swift:60]: Added

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift
Comment (igorkravchenko) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift:4]: @dukhovnyi @kuldar-daniel I think because most of the component styles are declared globally in the module, it makes sense to do the same for this style.

**Review by igorkravchenko — APPROVED** [review]
Good job 👍

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift
Comment (dukhovnyi) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayStyle.swift:4]: ok, let's do this way

File: GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift
Comment (dukhovnyi) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift:21]: but in case of a memory leak `deinit` method will not execute or do I miss something?

Comment (kuldar-daniel) [reply GliaWidgets/Component/OnHoldOverlay/OnHoldOverlayVisualEffectView.swift:21]: ah, you are correct. I guess I just saw online that it was common practice without really questioning it

**Review by dukhovnyi — APPROVED** [review]
Approve in advance, but this [comment](https://github.com/salemove/ios-sdk-widgets/pull/248#discussion_r830530796) has no answer.
Changes in this PR raised Swiftlint issue, so please fix it before merge.
One more suggestion, I'd like to propose to squash all commit in one to have a more flat history.

**Pull request merged**
